docs/docs/grpc-server.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: docs
+title: gRPC servers
+permalink: grpc/server/
+---
+
+# gRPC servers
+
+Mu-Haskell defines a generic notion of service and server that implements it. This generic server can then be used by `mu-grpc-server`, to provide a concrete implementation using a specific wire format.
+
+## Implementing the service
+
+Let's get back to the example we used in the [generic RPC section]({% link docs/rpc.md %}). In order to implement the corresponding service, you have to define the behavior of each method by means of a *handler*. You can use Haskell types for your handlers, given that you had previously declared that they can be mapped back and forth the schema types using `ToSchema` and `FromSchema`. For example, the following is a handler for the `SayHello` method in `Greeter`:
+
+```haskell
+sayHello :: (MonadServer m) => HelloRequest -> m HelloResponse
+sayHello (HelloRequest nm) = pure $ HelloResponse ("hi, " <> nm)
+```
+
+Notice the use of `MonadServer` in this case. This gives us the ability to:
+
+* Run arbitrary `IO` actions by using `liftIO`,
+* Return an error code by calling `serverError`.
+
+Being polymorphic here allows us to run the same server in multiple back-ends. Furthermore, by enlarging the set of abilities required for our monad `m`, we can [integrate with other libraries](transformer.md), including logging and resource pools.
+
+Since you can declare more than one method in a service, you need to join them into a `Server`. You do so by using `(:<|>:)` between each handler and ending the sequence with `H0`. In addition to the name of the service, `Server` has an additional parameter which records the types of the handlers. Since that list may become quite long, we can ask GHC to write it for us by using the `PartialTypeSignatures` extension and writing an underscore `_` in that position.
+
+```haskell
+{-# language PartialTypeSignatures #-}
+
+quickstartServer :: (MonadServer m) => ServerT QuickstartService m _
+quickstartServer = Server (sayHello :<|>: H0)
```

Comment:
This whole explanation about declaring services is outdated (if we merge your new PR), should I update it now or later? 😉

Reply:
My goal is to make this just the reorganization and GraphQL part. Then we merge and keep working on updating the rest.

Reply:
I've finally decided to include this into the docs.